### PR TITLE
Add filter to the feed group dialog to show only ungrouped subscriptions

### LINKED ISSUE #3377: A button to show only "ungrouped" channels in "select subscriptions"
#### Describe the feature you want

Many thanks for the new feature for organize the subscriptions!

At this moment I have ~270 subscriptions. The new feature is really useful but not enough to work with so many subscriptions. I will try to explain what's missing with an example.

I've a lot of subscriptions to channels about electronics. I would like to organize them in various groups, e.g. "Electronics-01", "Electronics-02", "Electronics-03", etc. I start with my favorites in the 01 group, and then filling the other groups with other channels. The problem is that I can't remember what channels are already in a different group.

So, my request is adding a button (or whatever you want) in the "select subscriptions" view to hide the channels which are already in other groups. Something like "show only ungrouped channels".

#### Is your feature request related to a problem? Please describe it

No.

#### How will you/everyone benefit from this feature?

It will help to users with a lot of subscriptions.

### PR DESCRIPTION
**Pull request by mauriciocolli**
#### What is it?
- [ ] Bug fix
- [x] Feature
- [ ] Code base improvement
- [ ] Meta improvement to the project

#### Description of the changes in your PR

Add a way to let user hide subscriptions from the subscription picker that are already included in any other group .

#### Fixes the following issue(s)
- Closes #3377

#### Relies on the following changes
- #3373

#### Testing apk

[feed-add-filter-sub-list.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/4840740/feed-add-filter-sub-list.apk.zip)

### CODE REVIEW

**Comment by denemier** [comment]
**NO don't**, i am using a channel related to multiple feeds.

I have local news channels in multiple feeds like
-news worldwide (includes local german channels)
-news only germany

but i am not against a button to filter it live for other users like you but i need this way

**Comment by juanfra684** [comment]
@denemier he is not changing the way the groups work. It's just a new option.

**Comment by MD77MD** [comment]
#3352
Add new subscribed channels to feed groups directly from button #3352

Describe the feature you want
The ability to add new subscribed channels to feed groups directly from the subscription button

Additional context
My suggestion is to add a drop down menu when pressing the subscription button containing all feed groups with a checkbox next to them. this is inspired from what youtube use to add videos to playlists

How will you/everyone benefit from this feature?
This would make adding new channels to groups immediate.. and would save so much effort especially if your adding the new channel to multiple groups. maybe try to add a channel to multiple groups to see what i mean.

**Comment by MD77MD** [comment]
#3353
New feed UI improved management #3353

Describe the feature you want
the ability to add subscribed channels to multiple existing feed groups directly from main subscription feed panel

Is your feature request related to a problem? Please describe it
Adding channels to a group require us to go through all subscribed channels.. which are alot, usually more than the feed groups

Additional context
My suggestion is to add an "add to group"
option within the -pressing and hold menu -
this option would open another drop-down menu containing all feed groups with a checkbox next to them. this is inspired from what youtube use to add videos to playlists

How will you/everyone benefit from this feature?
this would make adding and managing subscribed channels between feed groups a lot easier.

File: app/src/main/java/org/schabi/newpipe/database/subscription/SubscriptionDAO.kt [hunk 1]
```diff
@@ -22,10 +22,42 @@ abstract class SubscriptionDAO : BasicDAO<SubscriptionEntity> {
...
         SELECT * FROM subscriptions
+
         WHERE name LIKE '%' || :filter || '%'
+
+        ORDER BY name COLLATE NOCASE ASC
+        """)
+    abstract fun getSubscriptionsFiltered(filter: String): Flowable<List<SubscriptionEntity>>
+
+    @Query("""
+        SELECT * FROM subscriptions s
```
Comment (wb9688) [inline app/src/main/java/org/schabi/newpipe/database/subscription/SubscriptionDAO.kt:33]: Is it possible to omit `AS`? If so, I didn't know that.

Comment (mauriciocolli) [reply app/src/main/java/org/schabi/newpipe/database/subscription/SubscriptionDAO.kt:33]: Yes, keep in mind that Room won't let the app compile if any SQL statement is not correct. But it indeed is optional, either as a [column](https://www.sqlite.org/syntax/result-column.html) or a [table alias](https://www.sqlite.org/syntax/table-or-subquery.html).

If I remember correctly, the convention (at least in SQLite) is to use the `AS` keyword when aliasing a column and omit it when aliasing a table. 

###### Not so sure about that though.

**Comment by wb9688** [comment]
Could you rebase now that #3373 has been merged?

**Pull request merged**